cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -320,6 +324,15 @@ func (p *pod) CreateTask(ctx context.Context, req *task.CreateTaskRequest, s *sp
 		if !oci.IsJobContainer(s) {
 			return nil, errors.New("cannot create a normal process isolated container if the pod sandbox is a job container")
 		}
+		// Pass through some annotations from the pod spec that if specified will need to be made available
+		// to every container as well. Kubernetes only passes annotations to RunPodSandbox so there needs to be
+		// a way for individual containers to get access to these.
+		oci.SandboxAnnotationsPassThrough(
+			p.spec.Annotations,
+			s.Annotations,
```

Comment:
are these guaranteed to be non-nil?

Reply:
For CRI they always fill in some annotations, but maybe this would panic for ctr, nerdctl. other client etc. I'll add a guard

Reply:
Oh wait, this code would only get reached for a CRI scenario. Guess I can still bail if the pod spec annotations are nil 🤷‍♂️

Reply:
Yes, but the place this passthrough logic is placed in is only in our pod code which would only get reached for CRI